cassandra-cpp-driver 1.0.0 (new formula)

mpenick opened the pull request.

mpenick commented:
Formula for a C/C++ driver for Apache Cassandra. More information can be found here: http://datastax.github.io/cpp-driver/

DomT4 commented on Library/Formula/cassandra-cpp-driver.rb:
```diff
@@ -0,0 +1,20 @@
+require "formula"
```
This can be removed. Is redundant now.

DomT4 commented on Library/Formula/cassandra-cpp-driver.rb:
```diff
@@ -0,0 +1,20 @@
+  homepage "http://datastax.github.io/cpp-driver/"
+  url "https://github.com/datastax/cpp-driver/archive/1.0.0.tar.gz"
+  sha1 "087aa2cf00e3c6f0eb75bb0471b78d255ea94562"
+  version "1.0.0"
```
Unnecessary in this situation :), the Bot will be able to scan this from the URL.

DomT4 commented on Library/Formula/cassandra-cpp-driver.rb:
```diff
@@ -0,0 +1,20 @@
+  sha1 "087aa2cf00e3c6f0eb75bb0471b78d255ea94562"
+  version "1.0.0"
+
+  head "git://github.com:datastax/cpp-driver.git", :branch => "1.0"
```
https:// rather than git:// here please. Git tends to have problems behind corporate firewalls and the like, compared to 443.

DomT4 commented on Library/Formula/cassandra-cpp-driver.rb:
```diff
@@ -0,0 +1,20 @@
+  depends_on "libuv"
+
+  def install
+    mkdir 'build' do
```
Double quotes please :)

DomT4 commented on Library/Formula/cassandra-cpp-driver.rb:
```diff
@@ -0,0 +1,20 @@
+
+  def install
+    mkdir 'build' do
+      system "cmake", "-DCMAKE_BUILD_TYPE=RELEASE", "-DCASS_BUILD_STATIC=ON", "-DCMAKE_INSTALL_PREFIX:PATH=#{prefix}", ".."
```
The build type and prefix will be handled by ` std_cmake_args `. Switch to using ` std_cmake_args ` and then stick any other args afterwards, as seen in things like [here](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/json_spirit.rb).

DomT4 commented on Library/Formula/cassandra-cpp-driver.rb:
```diff
@@ -0,0 +1,20 @@
+      system "make", "install"
+    end
+  end
+end
```
Any chance of a ` test do ` block down here? The bot will moan at you without one :). Doesn't have to be hugely complicated. CMake's own test is [pretty simple](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/cmake.rb#L115-L118).

mpenick commented:
Updated. Thanks for the improvements. I updated everything except for the `test do` block because our current test infrastructure requires an extra large dependency: Boost.

mikemcquaid commented:
> Updated. Thanks for the improvements. I updated everything except for the test do block because our current test infrastructure requires an extra large dependency: Boost.

@mpenick You don't need to run your tests: you can just run a single invocation of a command to make sure it installed correctly.

DomT4 commented:
If still interested here please add a `test do` block and fix up anything else highlighted by `brew audit --strict`, Thanks!

DomT4 commented:
This seems to have gone dead. Closing consequently.

If you pick up interest later please feel free to resubmit as a new PR with the comments above addressed, Thanks!

Thank you for the submission!
